docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,451 @@
+
+#### expression(s) that constitutes the scanning range is hard to know
+
+To solve the problem of expression(s) that constitutes the scanning range is hard to know, we added a `RangeFilter` to the `KeyRange` to indicate the expression(s) used to construct the scan range.
+
+- **`RangeFilter`**: `RangeFilter` indicates which expression(s) the range is made up of. If `RangeFilter` is empty, it indicates a full table scan or full index scan. `RangeFilter` generally appears when the query involves an index range, when query the expressions in the `RangeFilter` form the scanned range from left to right.
+
+#### confusing naming of the operator
+
+To solve the problem of confusing the naming of the operator, we change the operator to the same name as TiDB. `IndexScan` has changed to `IndexLookUp`; `CoveringIndexScan` has changed to `IndexReader`; `TableScan` has changed to `TableReader`. The `PushDown Filter` has changed to `Selection`.
+
+- **`TableReader`**: Aggregates the data obtained by the underlying operator `TableRangeScan` in TiKV.
+- **`IndexReader`**: Aggregates the data obtained by the underlying operator `IndexRangeScan` in TiKV.
+- **`IndexLookUp`**: First aggregates the RowIDs (in TiKV) scanned by the first scan in the index. Then at the second scan in the table, accurately reads the data from TiKV based on these RowIDs. At the first scan in the index, there is  `IndexRangeScan` operator; at the second scan in the table, there is the `TableRowIDScan` operator.
+- **`Selection`**: The expression passed to COP/TiKV as selection expression without triggering a downgrade.
+
+### Code Design
+
+![image-20220628112756535](./imgs/phyical_plan_explain/code_flow_chart.png)
+
+## Test Design
+
+1. Table without cluster index
+
+   ```SQL
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   - `TableScan` with selection and without `RangeFilter`
+
+     ```sql
+     SELECT * FROM t1 where a>0 and b > 'aa'
+     ```
+
+2. Table with cluster index
+
+   TiDB version smaller than 5.0
+
+   ```sql
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) 
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   TiDB version bigger than 5.0
+
+   ```sql
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) clustered
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   - `TableScan` with selection and with `RangeFilter`
+
+     ```sql
+     SELECT * FROM t1 where a>0 and b > 'aa'
+     ```
+
+   - `TableScan` without selection and with `RangeFilter`
+
+     ```sql
+     SELECT * FROM t1 where a>0
+     ```
```

Comment:
do we need to test with selection and without `RangeFilter`? or it is impossible?

Reply:
add test